File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/GetConsentsBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton
import com.usercentrics.sdk.flutter.serializer.ConsentSerializer

internal class GetConsentsBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "getConsents"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        result.success(ConsentSerializer().serializeList(usercentrics.instance.getConsents()))
    }

}

File: ios/Classes/Bridge/GetConsentsBridge.swift
import Usercentrics

struct GetConsentsBridge : MethodBridge {
    let name: String = "getConsents"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        let consents = usercentrics.shared.getConsents()
        result(consents.map { $0.serialize() })
    }
}
